### PR TITLE
Orders refactoring

### PR DESCRIPTION
**Pull request by midas-myth**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207267409968783

### CODE REVIEW

**Review by onemicky — CHANGES_REQUESTED** [review]

File: src/components/Exchange/OrdersList.css [hunk 1]
```diff
@@ -1,77 +1,77 @@
-.Orders-list {
+/* .Orders-list {
```
[the remark this answers is not in the document]
Comment (midas-myth) [reply src/components/Exchange/OrdersList.css]: removed file, moved remaining rules to shared css

File: src/components/OrderEditorContainer/OrderEditorContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  const editingOrder = useSelector(selectEditingOrder);
+  const [, setEditingOrderKey] = useEditingOrderKeyState();
+  const [, setPendingTxns] = usePendingTxns();
+
+  if (!editingOrder) {
+    return null;
+  }
+
+  return (
+    <OrderEditor order={editingOrder} onClose={() => setEditingOrderKey(undefined)} setPendingTxns={setPendingTxns} />
```
[the remark this answers is not in the document]
Comment (midas-myth) [reply src/components/OrderEditorContainer/OrderEditorContainer.tsx]: done

File: src/components/Synthetics/OrderList/OrderList.tsx [hunk 3]
```diff
@@ -1,55 +1,96 @@
...
+  const orders = useFilteredOrders({
+    chainId,
+    account,
+    marketsDirectionsFilter: marketsDirectionsFilter,
+    orderTypesFilter: orderTypesFilter,
+  });
+
+  const onlySomeOrdersSelected =
+    selectedOrdersKeys && selectedOrdersKeys.length > 0 && selectedOrdersKeys.length < orders.length;
+  const areAllOrdersSelected = orders.length > 0 && orders.every((o) => selectedOrdersKeys?.includes(o.key));
```
[the remark this answers is not in the document]
Comment (midas-myth) [reply src/components/Synthetics/OrderList/OrderList.tsx]: done

File: src/components/Synthetics/OrderList/OrderList.tsx [hunk 4]
```diff
@@ -63,132 +104,210 @@ export function OrderList(p: Props) {
...
+                  isLarge={false}
+                  isSelected={selectedOrdersKeys?.includes(order.key)}
+                  onToggleOrder={() => onToggleOrder(order.key)}
+                  isCanceling={cancellingOrdersKeys.includes(order.key)}
+                  onCancelOrder={() => onCancelOrder(order.key)}
+                  positionsInfoData={positionsData}
+                  hideActions={hideActions}
+                  setRef={(el) => (orderRefs.current[order.key] = el)}
+                />
+              ))}
```
[the remark this answers is not in the document]
Comment (midas-myth) [reply src/components/Synthetics/OrderList/OrderList.tsx:210]: done the 1 way, i pass a (el, key)=>void handler

File: src/components/Synthetics/PositionItem/PositionItem.tsx [hunk 5]
```diff
@@ -769,3 +647,168 @@ export function PositionItem(p: Props) {
...
+  onOrdersClick,
+}: {
+  isSmall?: boolean;
+  positionKey: string;
+  onOrdersClick?: (key?: string) => void;
+}) {
+  const ordersWithErrors = usePositionOrdersWithErrors(positionKey);
+
+  const ordersErrorList = ordersWithErrors.filter(({ orderErrors }) => orderErrors.level === "error");
+  const ordersWarningsList = ordersWithErrors.filter(({ orderErrors }) => orderErrors.level === "warning");
```
[the remark this answers is not in the document]
Comment (midas-myth) [reply src/components/Synthetics/PositionItem/PositionItem.tsx]: done

File: src/components/Synthetics/PositionItem/PositionItem.tsx [hunk 6]
```diff
@@ -769,3 +647,168 @@ export function PositionItem(p: Props) {
...
+          <div className="flex items-center justify-between">
+            <PositionItemOrderText order={order} />
+            <FaAngleRight fontSize={16} className="ml-5" />
+          </div>
+        </Button>
+        <Button
+          variant="secondary"
+          className="!bg-slate-100 !bg-opacity-15 !p-6 hover:!bg-opacity-20 active:!bg-opacity-25"
+          onClick={() => {
+            setEditingOrderKey(order.key);
```
[the remark this answers is not in the document]
Comment (midas-myth) [reply src/components/Synthetics/PositionItem/PositionItem.tsx]: done

**Review by onemicky — DISMISSED** [review]

**Review by onemicky — APPROVED** [review]

**Pull request merged**